sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -89,6 +89,70 @@ pi_result piPluginGetLastError(char **message) {
   return ErrorMessageCode;
 }
 
+static cl_int getPlatformVersion(cl_platform_id plat,
+                                 OCLV::OpenCLVersion &version) {
+  cl_int ret_err;
+
+  size_t platVerSize;
+  ret_err =
+      clGetPlatformInfo(plat, CL_PLATFORM_VERSION, 0, nullptr, &platVerSize);
+
+  std::string platVer(platVerSize, '\0');
+  ret_err = clGetPlatformInfo(plat, CL_PLATFORM_VERSION, platVerSize,
+                              &platVer.front(), nullptr);
+
+  if (ret_err != CL_SUCCESS)
+    return ret_err;
+
+  version = OCLV::OpenCLVersion(platVer);
+  if (!version.isValid())
+    return CL_INVALID_CONTEXT;
+
+  return ret_err;
+}
+
+static cl_int getDeviceVersion(cl_device_id dev, OCLV::OpenCLVersion &version) {
+  cl_int ret_err;
+
+  size_t devVerSize;
+  ret_err = clGetDeviceInfo(dev, CL_DEVICE_VERSION, 0, nullptr, &devVerSize);
+
+  std::string devVer(devVerSize, '\0');
+  ret_err = clGetDeviceInfo(dev, CL_DEVICE_VERSION, devVerSize, &devVer.front(),
+                            nullptr);
+
+  if (ret_err != CL_SUCCESS)
+    return ret_err;
+
+  version = OCLV::OpenCLVersion(devVer);
+  if (!version.isValid())
+    return CL_INVALID_CONTEXT;
+
+  return ret_err;
+}
+
+static cl_int checkDeviceExtensions(cl_device_id dev,
+                                    std::vector<std::string> exts,
+                                    bool &supported) {
+  cl_int ret_err;
+
+  size_t extSize;
+  ret_err = clGetDeviceInfo(dev, CL_DEVICE_EXTENSIONS, 0, nullptr, &extSize);
+
+  std::string extStr(extSize, '\0');
+  ret_err = clGetDeviceInfo(dev, CL_DEVICE_EXTENSIONS, extSize, &extStr.front(),
+                            nullptr);
+
+  if (ret_err != CL_SUCCESS)
+    return ret_err;
+
+  supported = true;
+  for (std::string ext : exts)
+    supported &= (extStr.find(ext) != std::string::npos);
```

Comment:
Do not have strong opinion on that, but instead of having an additional argument we could return `CL_SUCCESS` if everything is supported and, for example `CL_INVALID_OPERATION`, if something is not.

Reply:
I agree, it would be nice to get rid of this extra argument. However, I think don't like the fact that it'd be kinda unclear whether `CL_INVALID_OPERATION` means that the extension wasn't found or a CL API call did fail.

Reply:
I'm OK with keeping the extra argument.